### PR TITLE
Remove resource template values when context is removed #8069 #modxbughunt

### PR DESCRIPTION
**Pull request by flowan**
### What does it do?
When a context is removed, also remove the resource template values that are linked to this context.

### Why is it needed?
Prevent a lot of overhead in the database.

### Related issue(s)/PR(s)
Related issue #8069

### CODE REVIEW

**Comment by gpsietzema** [comment]
#modxbughunt #3points to @Flowan

**Comment by pixelive** [comment]
Tested and works.

**Comment by gpsietzema** [comment]
#modxbughunt #1point to @pixelive

File: core/model/modx/processors/context/remove.class.php [hunk 1]
```diff
@@ -23,14 +23,31 @@ public function beforeRemove() {
     }
 
     public function afterRemove() {
+        /* Retrieve all resources from this context. */
+        $resources = $this->modx->getCollection('modResource',array(
```
Comment (Alroniks) [inline core/model/modx/processors/context/remove.class.php]: Maybe here make sense to use `getIterator` againt `getCollection `. It'll allow avoiding memory leak in a case when context had a lot of resources.

**Comment by Alroniks** [comment]
Looks nice and I am going to merge it, but could you have a look at my comment about `getIterator` @Flowan

File: core/model/modx/processors/context/remove.class.php (hunk 1 above)
Comment (Flowan) [inline core/model/modx/processors/context/remove.class.php]: Makes sense, i will update the code.

**Comment by Flowan** [comment]
Can be merged?

**Comment by sdrenth** [comment]
Also tested on 2.5.x branch and can confirm it works, so please merge?

**Pull request merged**